### PR TITLE
Fix Quorum

### PR DESCRIPTION
**Pull request by jiseongnoh**
## Proposed changes
This PR addresses a consensus instability issue in networks with 4x+1 validators. It updates the quorum calculation from the previous 2F+1 to math.Ceil(float64(2*valSet.Size())/3), aligning with QBFT requirements for enhanced network stability and Byzantine fault tolerance.

Closes https://github.com/klaytn/klaytn/issues/2023.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: consensus/istanbul/backend/engine.go [hunk 1]
```diff
@@ -357,7 +357,7 @@ func (sb *backend) verifyCommittedSeals(chain consensus.ChainReader, header *typ
 	}
 
 	// The length of validSeal should be larger than number of faulty node + 1
-	if validSeal <= 2*snap.ValSet.F() {
+	if validSeal < istanbulCore.RequiredMessageCount(snap.ValSet, chain.Config().IsCancunForkEnabled(header.Number)) {
```
Comment (hyeonLewis) [inline consensus/istanbul/backend/engine.go:360]: I think `2F()` can't be directly changed to `RequiredMessageCount` since it differs for small validators' cases. Is it okay?

Comment (hyeonLewis) [reply consensus/istanbul/backend/engine.go:360]: Oh, it was updated. But I think we still need to consider `n=6` case.

File: consensus/istanbul/core/core.go [hunk 2]
```diff
@@ -428,7 +435,10 @@ func PrepareCommittedSeal(hash common.Hash) []byte {
 }
 
 // Minimum required number of consensus messages to proceed
-func requiredMessageCount(valSet istanbul.ValidatorSet) int {
+func RequiredMessageCount(valSet istanbul.ValidatorSet, isAbcEnabled bool) int {
+	if isAbcEnabled {
+		return int(math.Ceil(float64(2*valSet.Size()) / 3))
```
[the remark this answers is not in the document]
Comment (jiseongnoh) [reply consensus/istanbul/core/core.go]: Thank you. reflected at e22d882c7dbfb34eb13c8f15e55aed181f38d0f9

File: consensus/istanbul/backend/engine.go (hunk 1 above)
Comment (jiseongnoh) [reply consensus/istanbul/backend/engine.go:360]: Let's try a non hard-fork-required one

**Review by blukat29 — APPROVED** [review]

**Review by hyeonLewis — APPROVED** [review]

File: consensus/istanbul/core/core.go [hunk 3]
```diff
@@ -428,20 +428,18 @@ func PrepareCommittedSeal(hash common.Hash) []byte {
 }
 
 // Minimum required number of consensus messages to proceed
-func requiredMessageCount(valSet istanbul.ValidatorSet) int {
+func RequiredMessageCount(valSet istanbul.ValidatorSet) int {
```
Comment (hyunsooda) [inline consensus/istanbul/core/core.go:431]: Is this counting change safe without HF? 
Assuming there are a total of `2F+1` nodes operational within the network, specifically denoting that `5` nodes are active among a total of `8` nodes.

In this scenario, where half of the nodes are responsible for calculating the necessary count of messages using the formula `2*valSet.F() + 1`, thereby anticipating `6` messages, while the other half calculates it using `int(math.Ceil(float64(2*size) / 3))`, expecting `5` messages.

Up to this point, the chain has been able to progress continuously. However, there emerges an issue wherein half of the updated nodes may not facilitate the forwarding of the chain. Does this not pose a potential risk or detriment associated with this change?

Comment (jiseongnoh) [reply consensus/istanbul/core/core.go:431]: The update to the quorum calculation without a hard fork is designed to be low-risk.

1. Cypress Network Stability: In the case of the Cypress mainnet, there are 31 committee nodes in operation, which aligns with the 3F+1 configuration. The quorum count will be consistent with both 2F+1 or Ceil(2N/3) formula. Therefore, the quorum calculation change will not pose a risk.
2. ServiceChain Network Resilience: ServiceChains with different numbers of validators might potentially experience a problem under extreme conditions. However, the problem will be solved as long as most nodes update the binary. This is acceptable considering the complexities of a hard fork.

In short, this method involves a minor relaxation of the BFT principle, a trade-off between an uninterrupted network and a quick update process.

Comment (jiseongnoh) [reply consensus/istanbul/core/core.go:431]: It is also noteworthy that under normal operating conditions, the number of validators participating in consensus typically surpass the quorum required by both the 2F+1 and Ceil(2*N/3) formulas. This surplus of committed seals ensures that discrepancies in different quorum calculation will not compromise the consensus process.

**Review by ian0371 — APPROVED** [review]

File: consensus/istanbul/core/core.go (hunk 3 above)
Comment (hyunsooda) [reply consensus/istanbul/core/core.go:431]: Got it thanks.

**Review by hyunsooda — APPROVED** [review]
